src/hex.rs

```diff
@@ -84,26 +84,43 @@ impl<'a> Iterator for HexIterator<'a> {
 
 /// Output hex into an object implementing `fmt::Write`, which is usually more
 /// efficient than going through a `String` using `ToHex`.
-pub fn format_hex<T: fmt::Write>(data: &[u8], mut fmt: T) -> fmt::Result {
-    for ch in data {
-        write!(fmt, "{:02x}", *ch)?;
+pub fn format_hex(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
+    let width = f.width().unwrap_or(2 * data.len());
+    for _ in (2 * data.len())..width {
+        f.write_str("0")?;
+    }
+    for ch in data.into_iter().take(width / 2) {
+        write!(f, "{:02x}", *ch)?;
+    }
+    if width < 2 * data.len() && width % 2 == 1 {
```

Comment:
Is it actually useful to support nibble-wise writing instead of byte-wise (I find it a little bit hard to reason about, but no big problem if you say it's useful)?

Reply:
The reason is only that bytewise would be confusing for a user, who would otherwise request `n` bytes and get `2n` characters.

I agree it's not really useful, the goal is just to be intuitive.

Reply:
Yeah, that seems like a worthwhile goal. Then I'll review this version.